### PR TITLE
Fix store info display

### LINKED ISSUE #3910: Bundle store frontend error
image

### PR DESCRIPTION
**Pull request by jzwang43**
### Reasons for making this change

Legacy bundles will not have bundle stores so we won't show the store info if API returns empty.

`BundleLocationListSchema(many=True).dump` doesn't seem to be working so I'm using `json.dumps` for now.

### Related issues

fixes #3910 

### Screenshots

For an old file upload:
image

New bundle:
image

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by epicfaace** [comment]
Looks good except for the `BundleLocationListSchema(many=True).dump(bundle_locations).data` part. Why didn't that work? did you get a specific error? Ideally, we would use that instead of using json.dump.

**Comment by jzwang43** [comment]
> 

The schema is defined incorrectly I believe. The error is `Unexpected Internal Error (Must have an `id` field). The administrators have been notified.`

**Comment by epicfaace** [comment]
Are you sure that's the error you get? BundleLocationListSchema doesn't have an id field defined (https://github.com/codalab/codalab-worksheets/blob/fix-store-info/codalab/rest/schemas.py#L247-L255) (only BundleStoreSchema does)

**Comment by jzwang43** [comment]
> Are you sure that's the error you get? BundleLocationListSchema doesn't have an id field defined (https://github.com/codalab/codalab-worksheets/blob/fix-store-info/codalab/rest/schemas.py#L247-L255) (only BundleStoreSchema does)

Yes I can confirm that's the error. If you look at the definition of the Schema class the BundleLocationListSchema extends, it requires an id field to be defined.

**Comment by epicfaace** [comment]
> > Are you sure that's the error you get? BundleLocationListSchema doesn't have an id field defined (https://github.com/codalab/codalab-worksheets/blob/fix-store-info/codalab/rest/schemas.py#L247-L255) (only BundleStoreSchema does)
> 
> Yes I can confirm that's the error. If you look at the definition of the Schema class the BundleLocationListSchema extends, it requires an id field to be defined.

Can you try reverting the BundleLocationListSchema(many=True).dump(bundle_locations).data -> json.dump change? I think the json change you did is causing the CI tests to fail. We should maybe get on a call to talk through this further to see what exactly the issue.

**Comment by jzwang43** [comment]
> > > Are you sure that's the error you get? BundleLocationListSchema doesn't have an id field defined (https://github.com/codalab/codalab-worksheets/blob/fix-store-info/codalab/rest/schemas.py#L247-L255) (only BundleStoreSchema does)
> > 
> > 
> > Yes I can confirm that's the error. If you look at the definition of the Schema class the BundleLocationListSchema extends, it requires an id field to be defined.
> 
> Can you try reverting the BundleLocationListSchema(many=True).dump(bundle_locations).data -> json.dump change? I think the json change you did is causing the CI tests to fail. We should maybe get on a call to talk through this further to see what exactly the issue.

If I revert the json.dumps change then the endpoint wouldn't work and this page will be broken.
The previous API I used was an old one and doesn't support multiple bundle locations.

File: codalab/model/bundle_model.py [hunk 1]
```diff
@@ -3090,6 +3090,7 @@ def get_bundle_locations(self, bundle_uuid: str) -> List[dict]:
             rows = connection.execute(
                 select(
                     [
+                        cl_bundle_store.c.id,
```
Comment (epicfaace) [inline codalab/model/bundle_model.py]: let's revert this file (which I changed in an attempt to fix the json issue)

**Review by epicfaace — APPROVED** [review]

**Pull request merged**